### ISSUE TITLE
etckeeper install error in build

### ISSUE THREAD

**Issue by wvengen** [turnkeylinux/tracker#354]
When building (rails) appliances, an error is shown when etckeeper is installed:

```
Setting up etckeeper (1.15) ...
Initialized empty Git repository in /etc/.git/

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'root@tkldev.(none)')
etckeeper commit failed; run it by hand
```

The appliance seems to build and run correctly though, and the git repository on a build appliance looks ok (`git log` shows `etckeeper <root@localhost>`). So perhaps not highest priority, but it would be nice to fix this.

A cause might be that `dnsdomainname` returns an empty value.
https://bugs.launchpad.net/ubuntu/+source/etckeeper/+bug/1267564
##

**Comment by JedMeister** [comment]
This is only because etckeeper is configured to use git in TKL and it seems that in Jessie git doesn't like the default 'user@appliance' email address anymore (it was fine for Wheezy...)

Should be fairly easy to workaround. Even if we preset it following the git example!?
````
  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"
````

**Comment by wvengen** [comment]
another similar issue
http://help.lockergnome.com/linux/Bug-583899-etckeeper-Git-user-email-settings--ftopict521752.html

I guess setting a global git configuration would work, though that would influence other uses of git as well. Perhaps this is something etckeeper packaging scripts should do on the repository after `git init`.

**Comment by JedMeister** [comment]
I agree that setting global is the wrong plan...

I propose that we set the etckeeper (`/etc`) git config to be something like this:
````
git config user.email "etckeeper@<appliance-name>.localhost"
git config user.name "etckeeper"
````
Thoughts? Alternative suggestions?

**Comment by wvengen** [comment]
Sounds good!

**Comment by JedMeister** [comment]
THere is a ton more work to do on v14.1 than I was expecting so seeing as this is non-fatal I'm repinning to v14.2 for now. Hope you don't mind.

**Comment by JedMeister** [comment]
I was about to shut this, but it was never really resolved, so I'm going to repin it to v15.0 and see what happens.

**Comment by JedMeister** [comment]
Actually, I am going to shut this. While it's a little ugly at build time, it's configured later on. And actually, that's probably better because it does it's first commit on firstboot, rather than at build time.

I just double checked that it was all working ok by launching a server, did basic config then left it. I just checked `/etc`:
````
root@core /etc# git log
commit ea87a5169496c0568190e9924d93e8f3b8862c6d
Author: etckeeper <root@localhost>
Date:   Thu Aug 3 06:48:06 2017 +0000

    daily autocommit

commit 5ea13095680cac24f2e630eaa3d7151dbc238261
Author: etckeeper <root@localhost>
Date:   Thu Aug 3 03:24:04 2017 +0000

    saving uncommitted changes in /etc prior to apt run

commit c475e6409e34b9e5432b15ba47ba29bcfeb7b95e
Author: etckeeper <root@localhost>
Date:   Thu Aug 3 03:20:52 2017 +0000

    initial commit
````
